### PR TITLE
[WIP] Revert templates to 1.x

### PR DESCRIPTION
**Pull request by javiercn**

### CODE REVIEW

**Comment by javiercn** [comment]
@mlorbetske @danroth27

**Comment by danroth27** [comment]
@HaoK

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Controllers/ManageController.cs [hunk 1]
```diff
@@ -274,19 +300,23 @@ public IActionResult SetPassword()
...
 
+        //
+        // POST: /Manage/LinkLogin
         [HttpPost]
         [ValidateAntiForgeryToken]
         public async Task<IActionResult> LinkLogin(string provider)
         {
             // Clear the existing external cookie to ensure a clean login process
-            await HttpContext.SignOutAsync(IdentityCookieOptions.ExternalScheme);
+            await HttpContext.Authentication.SignOutAsync(_externalCookieScheme);
```
Comment (HaoK) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Controllers/ManageController.cs]: This change should stay, this is an Auth 2.0 change

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Controllers/ManageController.cs [hunk 2]
```diff
@@ -306,7 +336,7 @@ public IActionResult SetPassword()
             {
                 message = ManageMessageId.AddLoginSuccess;
                 // Clear the existing external cookie to ensure a clean login process
-                await HttpContext.SignOutAsync(IdentityCookieOptions.ExternalScheme);
+                await HttpContext.Authentication.SignOutAsync(_externalCookieScheme);
```
Comment (HaoK) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Controllers/ManageController.cs]: This also needs to stay

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Controllers/ManageController.cs [hunk 3]
```diff
@@ -264,8 +291,7 @@ public IActionResult SetPassword()
                 return View("Error");
             }
             var userLogins = await _userManager.GetLoginsAsync(user);
-            var schemes = await _signInManager.GetExternalAuthenticationSchemesAsync();
-            var otherLogins = schemes.Where(auth => userLogins.All(ul => auth.Name != ul.LoginProvider)).ToList();
+            var otherLogins = _signInManager.GetExternalAuthenticationSchemes().Where(auth => userLogins.All(ul => auth.AuthenticationScheme != ul.LoginProvider)).ToList();
```
Comment (HaoK) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Controllers/ManageController.cs]: Needs to stay

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Controllers/ManageController.cs [hunk 4]
```diff
@@ -1,43 +1,46 @@
...
         private readonly SignInManager<ApplicationUser> _signInManager;
+        private readonly string _externalCookieScheme;
         private readonly IEmailSender _emailSender;
         private readonly ISmsSender _smsSender;
         private readonly ILogger _logger;
 
         public ManageController(
           UserManager<ApplicationUser> userManager,
           SignInManager<ApplicationUser> signInManager,
+          IOptions<IdentityCookieOptions> identityCookieOptions,
```
Comment (HaoK) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Controllers/ManageController.cs]: Remove

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Controllers/ManageController.cs [hunk 5]
```diff
@@ -1,43 +1,46 @@
...
           IEmailSender emailSender,
           ISmsSender smsSender,
           ILoggerFactory loggerFactory)
         {
             _userManager = userManager;
             _signInManager = signInManager;
+            _externalCookieScheme = identityCookieOptions.Value.ExternalCookieAuthenticationScheme;
```
Comment (HaoK) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Controllers/ManageController.cs]: Remove

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Controllers/AccountController.cs [hunk 6]
```diff
@@ -1,90 +1,529 @@
...
+        // GET: /Account/Login
         [HttpGet]
-        public IActionResult SignIn()
+        [AllowAnonymous]
+        public async Task<IActionResult> Login(string returnUrl = null)
         {
-            return Challenge(
-                new AuthenticationProperties { RedirectUri = "/" }, OpenIdConnectDefaults.AuthenticationScheme);
+            // Clear the existing external cookie to ensure a clean login process
+            await HttpContext.Authentication.SignOutAsync(_externalCookieScheme);
```
Comment (HaoK) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Controllers/AccountController.cs]: await HttpContext.SignOutAsync(IdentityConstants.ExternalScheme)

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Controllers/ManageController.cs (hunk 1 above)
Comment (HaoK) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Controllers/ManageController.cs]: Constant needs to be updated to: await HttpContext.SignOutAsync(IdentityConstants.ExternalScheme)

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Controllers/ManageController.cs (hunk 2 above)
Comment (HaoK) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Controllers/ManageController.cs]: await HttpContext.SignOutAsync(IdentityConstants.ExternalScheme)

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Data/Migrations/ApplicationDbContextModelSnapshot.cs [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+        {
+            modelBuilder
+#if (UseLocalDB)
+                .HasAnnotation("ProductVersion", "1.0.0-rc3")
+                .HasAnnotation("SqlServer:ValueGenerationStrategy", SqlServerValueGenerationStrategy.IdentityColumn);
+#else
+                .HasAnnotation("ProductVersion", "1.0.2");
+#endif
+
+            modelBuilder.Entity("Microsoft.AspNetCore.Identity.EntityFrameworkCore.IdentityRole", b =>
```
Comment (HaoK) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Data/Migrations/ApplicationDbContextModelSnapshot.cs]: Need to remove the EntityFrameworkCore in the namespace for all of these entities in the migration, i.e.:

"Microsoft.AspNetCore.Identity.IdentityRole"

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Extensions/AzureAd/AzureAdOptions.cs [hunk 8]
```diff
@@ -1,12 +0,0 @@
-﻿namespace Microsoft.AspNetCore.Authentication.Extensions
-{
-    public class AzureAdOptions
```
Comment (HaoK) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Extensions/AzureAd/AzureAdOptions.cs]: @danroth27 are we keeping these?

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Models/ManageViewModels/ManageLoginsViewModel.cs [hunk 9]
```diff
@@ -2,15 +2,15 @@
...
 
-namespace Company.WebApplication1.Identity.Models.ManageViewModels
+namespace Company.WebApplication1.Models.ManageViewModels
 {
     public class ManageLoginsViewModel
     {
         public IList<UserLoginInfo> CurrentLogins { get; set; }
 
-        public IList<AuthenticationScheme> OtherLogins { get; set; }
+        public IList<AuthenticationDescription> OtherLogins { get; set; }
```
Comment (HaoK) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Models/ManageViewModels/ManageLoginsViewModel.cs]: Keep this change

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Models/ManageViewModels/ManageLoginsViewModel.cs [hunk 10]
```diff
@@ -2,15 +2,15 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
-using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Http.Authentication;
```
Comment (HaoK) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Models/ManageViewModels/ManageLoginsViewModel.cs]: Keep this change

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Program.cs [hunk 11]
```diff
@@ -3,23 +3,25 @@
...
     public class Program
     {
         public static void Main(string[] args)
         {
-            BuildWebHost(args).Run();
-        }
-
-        public static IWebHost BuildWebHost(string[] args) =>
-            WebHost.CreateDefaultBuilder(args)
+            var host = new WebHostBuilder()
```
Comment (HaoK) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Program.cs]: We still want to use the WebHost stuff right?

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Startup.cs [hunk 12]
```diff
@@ -80,10 +115,72 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env)
 
             app.UseStaticFiles();
 
-#if (OrganizationalAuth || IndividualAuth)
-            app.UseRewriter(new RewriteOptions().AddIISUrlRewrite(env.ContentRootFileProvider, "urlRewrite.config"));
+#if (IndividualAuth)
+            app.UseIdentity();
```
Comment (HaoK) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Startup.cs]: UseIdentity => UseAuthentication()

**Pull request merged**